packages/client/src/query/aggregations/GroupByClause.ts

```diff
@@ -25,11 +25,52 @@ export type GroupByClause<
 };
 export type StringGroupByValue = "exact" | { exactWithLimit: number };
 
-export type GroupByRange = [number, number];
+export type GroupByRange<T> = [T, T];
+export type DurationGroupBy = [number, keyof typeof TimeUnitMapping];
 
 export type NumericGroupByValue = "exact" | { exactWithLimit: number } | {
   fixedWidth: number;
-} | { ranges: GroupByRange[] };
+} | { ranges: GroupByRange<number>[] };
+
+export type TimestampGroupByValue =
+  | "exact"
+  | { ranges: GroupByRange<string>[] }
+  | { duration: DurationGroupBy };
+
+export type TimeUnit =
+  | "SECONDS"
+  | "MINUTES"
+  | "HOURS"
+  | "DAYS"
+  | "WEEKS"
+  | "MONTHS"
+  | "YEARS"
+  | "QUARTERS";
+
+export const TimeUnitMapping = {
```

Comment:
Was trying to emulate the flexibility we gave here https://github.com/palantir/osdk-ts/blob/012aae96dcbb063e8d6ca5f1efb4e74985da97d5/packages/client/src/query/WhereClause.ts#L63C1-L80C20 for the within clause, but may have added way too much, can definitely cut down